datafusion/core/tests/window.rs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Tests for window queries
+use std::sync::Arc;
+
+use arrow_array::RecordBatch;
+use arrow_schema::{DataType, Field, Schema};
+use datafusion::prelude::SessionContext;
+use datafusion_common::{assert_batches_eq, Result, ScalarValue};
+use datafusion_execution::config::SessionConfig;
+use datafusion_expr::{WindowFrame, WindowFrameBound, WindowFrameUnits};
+use datafusion_physical_expr::expressions::{col, NthValue};
+use datafusion_physical_expr::window::{BuiltInWindowExpr, BuiltInWindowFunctionExpr};
+use datafusion_physical_plan::memory::MemoryExec;
+use datafusion_physical_plan::windows::{BoundedWindowAggExec, PartitionSearchMode};
+use datafusion_physical_plan::{collect, get_plan_string, ExecutionPlan};
+
+// Tests NTH_VALUE(negative index) with memoize feature.
+// To be able to trigger memoize feature for NTH_VALUE we need to
+// - feed BoundedWindowAggExec with batch stream data.
+// - Window frame should contain UNBOUNDED PRECEDING.
+// It hard to ensure these conditions are met, from the sql query.
```

Comment:
I think this test would more naturally belong in the same module as `BoundedWindowAggExec` as it uses pretty low level APIs (i wouldn't really expect any users to ever write this kind of query)

I suggest:
1. Change this test to use the LogicalPlan or DataFrame APIs instead
2. Move the test into https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-plan/src/windows/bounded_window_agg_exec.rs

Reply:
Moved under bounded_window_agg_exec file as you suggest. 👍